tenant-file/portal-app/src/component/admin.tsx

```diff
@@ -0,0 +1,257 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { getToken } from "./firebase";
+import firebase from "firebase";
+import {
+  Row,
+  Container,
+  Col,
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  FormGroup,
+  Label,
+  Input,
+} from "reactstrap";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+
+interface UserRecord {
+  uid: string;
+  email: string;
+  claims: Record<string, boolean>;
+  displayName: string;
+}
+
+const Admin: React.FC = () => {
+  const baseUrl =
+    process.env.NODE_ENV === "production"
+      ? "https://tenant-file-api-zmzadnnc3q-uc.a.run.app"
+      : "http://localhost:8080";
+
+  let [users, setUsers] = useState<UserRecord[]>([]);
+
+  const [modal, setModal] = useState(false);
+  const [deleteModalOpen, setDeleteModalOpen] = useState(false);
+
+  const [userBeingModified, setUserBeingModified] = useState<UserRecord | null>(
+    null
+  );
+  const [editMode, setEditMode] = useState<"edit" | "create">("create");
+
+  const toggleEditModal = () => setModal(!modal);
+  const toggleDeleteModal = () => {
+    if (deleteModalOpen) setUserBeingModified(null);
+    setDeleteModalOpen(!deleteModalOpen);
+  };
+
+  useEffect(() => {
+    const func = async () => {
+      const token = await getToken();
+      const usersResponse = await axios
+        .get(`${baseUrl}/users/list`, {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        })
+        .then((x) => x.data);
+      console.log("RESPONSE", usersResponse);
+      setUsers(usersResponse.users);
+    };
+    func();
+  }, []);
+
+  return (
+    <>
+      <Modal isOpen={modal} toggle={toggleEditModal}>
+        <ModalHeader toggle={toggleEditModal}>Modal title</ModalHeader>
+
+        <Formik
```

Comment:
I'm really not a React expert but I think for clarity/organization sake it might be useful to break out this `<Formik>` into its own component and create a "pages" folder for the Admin and other page elements that are routed to.